### PR TITLE
🔥 Remove Java 8 source/target compatibility from Android snippets & update Android permission examples

### PR DESCRIPTION
**Pull request by phatblat**
Changes in getditto/ditto#5882 will make it no longer necessary for apps to downgrade their `sourceCompatibility` and `targetCompatibility` in order to use Ditto on Android.

## Java Version

```gradle
android {
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }
}
```

## Permissions

Also rolled getditto/ditto#5259 into this work. Updated android permissions to the following:

```xml
<uses-permission android:name="android.permission.BLUETOOTH"
    android:maxSdkVersion="30" />
<uses-permission android:name="android.permission.BLUETOOTH_ADMIN"
    android:maxSdkVersion="30" />
<uses-permission android:name="android.permission.BLUETOOTH_ADVERTISE"
    tools:targetApi="s" />
<uses-permission android:name="android.permission.BLUETOOTH_CONNECT"
    tools:targetApi="s" />
<uses-permission android:name="android.permission.BLUETOOTH_SCAN"
    android:usesPermissionFlags="neverForLocation"
    tools:targetApi="s" />
<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"
    android:maxSdkVersion="32" />
<uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"
    android:maxSdkVersion="30" />
<uses-permission android:name="android.permission.INTERNET" />
<uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
<uses-permission android:name="android.permission.CHANGE_NETWORK_STATE" />
<uses-permission android:name="android.permission.CHANGE_WIFI_MULTICAST_STATE" />
<uses-permission android:name="android.permission.CHANGE_WIFI_STATE" />
<uses-permission android:name="android.permission.NEARBY_WIFI_DEVICES"
    android:usesPermissionFlags="neverForLocation"
    tools:targetApi="tiramisu" />
```

This set of permissions should work for any project, even if they aren't targeting Tiramisu (Android 13, API 33).

- `maxSdkVersion` makes that permission apply to everything up to that API version
- `targetApi` makes that permission only apply when running on a device with at least that API version

## API/SDK Levels

Updated our docs **lowering** our minimum Android version and adding our required **compile** SDK (these are unchanged)

> Android 6 (`minSdk` = 23) and above with `compileSdk` >= 31.

Wait to merge/deploy this until the changes in the referenced PR is released.

### CODE REVIEW

**Comment by okdistribute** [comment]
Awesome!

**Review by thombles — APPROVED** [review]
LGTM 👍 

I dimly recall that the Java 8 source compatibility was originally put in the gradle script for the Android dnssd library, and it would crash on transport startup if you omitted it. I haven't followed this closely but I'm assuming that's no longer a concern or we'd have noticed.

File: docs/android/installation.mdx [hunk 1]
```diff
@@ -1,29 +1,30 @@
...
 :::
 
-|  | LAN | WiFi Direct | Bluetooth | WebSockets |
-|---|---| ---|--- | ---|
-Android | ✓ | ✓ | ✓| ✓ |
-Desktop | x | x | x | x |
+|         | LAN        | WiFi Direct | Bluetooth | WebSockets |
+| ------- | ---------- | ----------- | --------- | ---------- |
+| Android | ✓          | ✓           | ✓         | ✓          |
+| Desktop | Contact us | x           | x         | x          |
```
[the remark this answers is not in the document]
Comment (phatblat) [reply docs/android/installation.mdx:22]: Much better 👌🏻

[the remark this answers is not in the document]
Comment (phatblat) [reply docs/android/installation.mdx]: Should we add WiFi Aware to this table?

[the remark this answers is not in the document]
Comment (okdistribute) [reply docs/android/installation.mdx]: Is WiFi Direct the same as WiFi Aware? Maybe just change the name?

[the remark this answers is not in the document]
Comment (okdistribute) [reply docs/android/installation.mdx]: Most people just want P2P Wifi (is that WiFi Aware?) We should just mention that one, then.

[the remark this answers is not in the document]
Comment (phatblat) [reply docs/android/installation.mdx]: On Android, both WiFi Direct and WiFi Aware are "P2P WiFi"

**Comment by phatblat** [comment]
Thanks @okdistribute!

**Pull request merged**